src/app/modules/item/components/chapter-children/chapter-children.component.html

```diff
@@ -16,34 +16,85 @@
       This chapter has no content visible to you, so you can't validate it for now.
     </p>
 
-    <ul class="data-list-container activity" *ngIf="data.children.length > 0">
-      <li *ngFor="let item of data.children; index as ln">
-      <span class="activity-progress">
-        <alg-score-ring
-          *ngIf="(!item.watchedGroup && item.result) || (item.watchedGroup && item.watchedGroup.avgScore !== undefined)"
-          [diameter]="30"
-          [currentScore]="(item.watchedGroup ? item.watchedGroup.avgScore : item.result?.score) ?? 0"
-          [isValidated]="(item.watchedGroup ? item.watchedGroup.allValidated : item.result?.validated) ?? false"
-          [bestScore]="item.watchedGroup ? (item.watchedGroup.avgScore ?? 0) : item.bestScore"
-        ></alg-score-ring>
-      </span>
-      <span
-        class="activity-content"
-        (click)="click(item)"
-        [ngClass]="{ locked: item.isLocked }"
+    <ng-container *ngIf="itemData && itemData.item.childrenLayout === 'Grid'; else listSection">
+      <div
+        class="grid-container"
+        [ngClass]="{ 'full-frame': (fullFrame$ | async)?.active }"
+        *ngIf="data.children.length > 0"
       >
-        <a class="activity-title alg-link">
-          {{ ln + 1 }}. {{ item.string.title }}
-        </a>
-        <span *ngIf="item.isLocked" class="activity-status">
-          <i *ngIf="item.isLocked" class="fa fa-lock"></i>
-        </span>
-        <alg-state-widget
-          [type]="item.category"
-        ></alg-state-widget>
-      </span>
-      </li>
-    </ul>
+        <div class="grid-item" *ngFor="let item of data.children; let i = index">
+          <div class="grid-card" (click)="click(item)">
+            <div class="grid-card-title text-dots">{{ i + 1 }}. {{ item.string.title }}</div>
+            <div class="grid-card-image-section">
+              <div class="grid-card-lock-status">
+                <i
+                  class="grid-card-lock-status-icon fa"
+                  [ngClass]="{
+                    'fa-lock': item.isLocked,
+                    'fa-key': !item.isLocked
+                  }"
+                ></i>
+              </div>
+              <div class="grid-card-lock-bg" *ngIf="item.isLocked"></div>
+              <div
+                class="grid-card-image"
+                [style.background-image]="'url(' + item.string.imageUrl + ')'"
+                *ngIf="item.string.imageUrl; else defaultImage"
+              ></div>
+              <ng-template #defaultImage>
+                <div
+                  class="grid-card-image"
+                  [style.background-image]="'url(assets/images/chapter-default.jpg)'"
```

Comment:
shouldn't it be in the CSS so that it is treated correctly by webpack?

Reply:
Fixed.

I guess the picture won't be visible in test cases as it tries to get it from https://assets.algorea.org/assets/images/chapter-default.jpg

But the picture has been deployed to branch:

https://assets.algorea.org/branch/feature/grid-view-for-chapter/en/assets/images/chapter-default.jpg 

So I assume it will be displayed after merge